src/orchard/java.clj

```diff
@@ -268,11 +268,13 @@
   implemention. If the member is an instance member, `this` is prepended to its
   arglists."
   [class member]
-  (let [c (->> (class-info class)
-               (iterate (comp class-info :super))
-               (take-while identity)
-               (filter #(get-in % [:members member]))
-               (first))]
+  (let [c (loop [next-class class]
```

Comment:
As this code is not idiomatic it should definitely include some comments to discourage people from optimizing it in the other direction. :-)